Sources/Starknet/Data/Execution.swift

```diff
@@ -21,7 +23,47 @@ public struct StarknetCall: Codable, Equatable {
     }
 }
 
-public struct StarknetExecutionParams {
+public struct StarknetExecutionParamsV3 {
+    public let nonce: Felt
+    public let resourceBounds: StarknetResourceBoundsMapping
+    public let tip: UInt64AsHex
+    public let paymasterData: StarknetPaymasterData
+    public let accountDeploymentData: StarknetAccountDeploymentData
+    public let nonceDataAvailabilityMode: StarknetDAMode
+    public let feeDataAvailabilityMode: StarknetDAMode
+
+    public init(nonce: Felt, l1ResourceBounds: StarknetResourceBounds) {
+        self.nonce = nonce
+        // As of Starknet 0.13, most of v3 fields have hardcoded values.
+        self.resourceBounds = StarknetResourceBoundsMapping(l1Gas: l1ResourceBounds)
+        self.tip = .zero
+        self.paymasterData = []
+        self.accountDeploymentData = []
+        self.nonceDataAvailabilityMode = .l1
+        self.feeDataAvailabilityMode = .l1
+    }
+}
```

Comment:
Should we add separate `DeployAccount` and `Invoke` params?

Currently, the only difference is `accountDeploymentData`. But once it's not hardcoded anymore, I don't know how problematic and frequent is the situation where user sets `accountDeploymentData` and passes it to `signDeployAccountV3`. It shouldn't break anything, it will be just ignore, but that is also not ideal.

If we wanna do it, we should probably do it now to avoid extra breaking changes later. 

My reason for not doing it right away is existence of optional params. It seems to me that having `StarknetInvokeParamsV3`, `StarknetOptionalInvokeParamsV3` , `StarknetDeployAccountParamsV3`,  `StarknetOptionalDeployAccountParamsV3` alongside some extra v1 params is not that intuitive.

Wdyt?

Reply:
let's have them separated, to be a bit more type-safe. Probably makes more sense for the future as well.